src/report.rs

```diff
@@ -407,12 +417,112 @@ pub fn error_row(error: &metrics::GooseErrorMetricAggregate) -> String {
     )
 }
 
+/// Build a requests per second graph.
 pub fn graph_rps_template(
     rps: &[(String, u32)],
     starting: Option<DateTime<Local>>,
     started: Option<DateTime<Local>>,
     stopping: Option<DateTime<Local>>,
     stopped: Option<DateTime<Local>>,
+) -> String {
+    graph_template(
+        "graph-rps",
+        "Requests #",
+        rps,
+        starting,
+        started,
+        stopping,
+        stopped,
+    )
+}
+
+/// Build an errors per second graph.
+pub fn graph_eps_template(
+    eps: &[(String, u32)],
+    starting: Option<DateTime<Local>>,
+    started: Option<DateTime<Local>>,
+    stopping: Option<DateTime<Local>>,
+    stopped: Option<DateTime<Local>>,
+) -> String {
+    graph_template(
+        "graph-eps",
+        "Errors #",
+        eps,
+        starting,
+        started,
+        stopping,
+        stopped,
+    )
+}
+
+/// Build an average response time graph.
+pub fn graph_average_response_time_template(
+    response_times: &[(String, u32)],
+    starting: Option<DateTime<Local>>,
+    started: Option<DateTime<Local>>,
+    stopping: Option<DateTime<Local>>,
+    stopped: Option<DateTime<Local>>,
+) -> String {
+    graph_template(
+        "graph-avg-response-time",
+        "Response time [ms]",
+        response_times,
+        starting,
+        started,
+        stopping,
+        stopped,
+    )
+}
+
+/// Build a users per second graph.
+pub fn graph_users_per_second_template(
+    users: &[(String, usize)],
+    starting: Option<DateTime<Local>>,
+    started: Option<DateTime<Local>>,
+    stopping: Option<DateTime<Local>>,
+    stopped: Option<DateTime<Local>>,
+) -> String {
+    graph_template(
+        "graph-active-users",
+        "Active users #",
+        users,
+        starting,
+        started,
+        stopping,
+        stopped,
+    )
+}
+
+/// Build a tasks per second graph.
+pub fn graph_tasks_per_second_template<T: Serialize>(
+    tasks: &[(String, T)],
+    starting: Option<DateTime<Local>>,
+    started: Option<DateTime<Local>>,
+    stopping: Option<DateTime<Local>>,
+    stopped: Option<DateTime<Local>>,
+) -> String {
+    graph_template(
+        "graph-tps",
+        "Tasks #",
+        tasks,
+        starting,
+        started,
+        stopping,
+        stopped,
+    )
+}
+
+/// Helper function to build HTML charts powered by the
+/// [ECharts](https://echarts.apache.org) library.
+#[allow(clippy::too_many_arguments)]
```

Comment:
I always aim to restructure how I call the function instead of selectively ignoring Clippy. Perhaps we can create an object to pass in, for example?

Reply:
Makes sense. Done. 

I ended up moving the function that generates markup onto the struct as well. This looks much nicer to me now.